raco/rules.py

```diff
@@ -832,6 +832,17 @@ def do_transfer(op):
             # Need to Shuffle
             op.input = algebra.Shuffle(op.input, group_fields)
 
+    @staticmethod
+    def check_no_shuffle(op):
+        """Check if no shuffle is needed"""
+
+        # Get an array of position references to columns in the child scheme
+        child_scheme = op.input.scheme()
+        group_fields = [expression.toUnnamed(ref, child_scheme)
+                        for ref in op.grouping_list]
+        return len(group_fields) > 0 \
```

Comment:
Nitpick: I prefer to avoid backslash continuations in favor of grouping parentheses.